metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -569,7 +569,7 @@ public Partition alterPartition(final RawStore msdb, Warehouse wh, final String
           LOG.error("Revert the data move in renaming a partition.");
           try {
             if (destFs.exists(destPath)) {
-              wh.renameDir(destPath, srcPath);
+              wh.renameDir(destPath, srcPath, false);
```

Comment:
here we are using the same rename method for reverting back CM changes, seems to be better to have a different method for it, so we dont try to figure out why there is a rename in CM without recycle option. this is only to be called in case of reverting back.